### ISSUE TITLE
Feature: convert docs to pdf

### ISSUE THREAD

**Issue by andrewwippler** [bsdz/google-drive-remarkable-sync#9]
I just received my remarkable 2 and am using it to replace paper (I hate printing things). I set up this app script and was kind of disappointed that this only syncs PDF and ePUB files.

Considering GDrive has PDF conversion built in [by accessing the blob object as PDF](https://developers.google.com/apps-script/reference/base/blob#getAs(String)), would it be as simple as editing https://github.com/bsdz/google-drive-remarkable-sync/blob/master/Synchronizer.js#L158 to grab it as a PDF? Any other gotchas (such as https://github.com/bsdz/google-drive-remarkable-sync/blob/master/Synchronizer.js#L243) that I need to be aware of when implementing this feature?

I use GDrive as my word processor and having updated notes auto synced to my remarkable will be huge - even if it is only one way.

**Comment by bsdz** [comment]
> Considering GDrive has PDF conversion built in [by accessing the blob object as PDF](https://developers.google.com/apps-script/reference/base/blob#getAs(String)), would it be as simple as editing https://github.com/bsdz/google-drive-remarkable-sync/blob/master/Synchronizer.js#L158 to grab it as a PDF? Any other gotchas (such as https://github.com/bsdz/google-drive-remarkable-sync/blob/master/Synchronizer.js#L243) that I need to be aware of when implementing this feature?

I feel this would be a nice feature to add and, tbh, I wasn't even aware of GDrive's built-in blob conversion until now. You seem to have the gist of what's going on in the Synchronizer. There might be some additional logic needed since the original file names might be like "foo.doc" whereas the targets will be like "foo.pdf".

**Comment by andrewwippler** [comment]
Please review #11 and comment on the code.

**Comment by andrewwippler** [comment]
After a few days of use, I can see the changes being seen on the script, however, I only get cached copies on my tablet/phone app. Any ideas on why the server is not being updated with the new version?

**Comment by bsdz** [comment]
Sorry I've not had a chance to look at it properly yet. I think my initial thoughts were it looked mostly fine although I was uncertain about the need to capture and store last runtime. I doubt that would cause the files not to update. I do recall if one doesn't upload the correct meta data then the server might ignore the corresponding file.

**Comment by andrewwippler** [comment]
No worries about not looking at it - I realize I did it over the holiday and life is busy :)

I stored the last run time so that the sync could select only updated files and skip over unchanged files. It was quite the painful process for my small 100 file workload to have it sync over all files every script run.

I am still doing a bit of debugging on my end. At first glance, it seems if I delete the file on the remarkable, it goes to the trash folder. A new sync moves the trashed file to its previous location without updating its contents. I am going to take a closer look.
